charts/datadog/templates/secret-application-key.yaml

```diff
@@ -7,6 +7,9 @@ metadata:
   namespace: {{ .Release.Namespace }}
   labels:
 {{ include "datadog.labels" . | indent 4 }}
+{{- if .Values.datadog.secretAnnotations }}
+  annotations: {{ toYaml .Values.datadog.secretAnnotations | nindent 4 }}
```

Comment:
```suggestion
  annotations:
    {{ toYaml .Values.datadog.secretAnnotations | nindent 4 }}
```

Reply:
I think @Izzette suggestion is needed.

Reply:
No I made a mistake, I read `indent` not `nindent`.  `nindent` will commence with a newline and as such not be problematic in formatting.  This will leave a trailing space after the colon, however the enhanced readability of the template is likely more important.

Reply:
IMO, It is still needed because the line should be under the "annotations:" line

Reply:
from what I understand these two forms are equivalent and will generate the same manifest all the time.
```yaml
foo: {{ toYaml .Values.bar | nindent X }}
```
and
```yaml
foo: 
  {{ toYaml .Values.bar | indent X }}
```

Generating with @maximumG's code I produce this secret manifest:
```sh
helm template \
    --values examples/datadog/agent_basic_values.yaml \
    --set-json datadog.secretAnnotations='{"annota.ti.on/foobar": "value"}' \
    --set datadog.clusterName=my-cluster charts/datadog |
  ruby -r yaml -e '
    YAML.load_stream($stdin) do |manifest|
      if manifest["kind"] == "Secret"
        puts YAML.dump(manifest)
      end
    end'
```
```yaml
---
apiVersion: v1
kind: Secret
metadata:
  name: release-name-datadog-cluster-agent
  namespace: default
  labels:
    helm.sh/chart: datadog-3.7.2
    app.kubernetes.io/name: release-name-datadog
    app.kubernetes.io/instance: release-name
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/version: '7'
  annotations:
    annota.ti.on/foobar: value
type: Opaque
data:
  token: WDlIMktIZXI3cm95SVdOcUxiQmc5eldmaXlMRHpPTzk=
```

Reply:
Right, I didn't saw the `nindent` instead on `indent`